scwconfig/config.go

```diff
@@ -0,0 +1,327 @@
+package scwconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Environment variables
+const (
+	// Up-to-date
+	scwConfigPathEnv            = "SCW_CONFIG_PATH"
+	scwAccessKeyEnv             = "SCW_ACCESS_KEY"
+	scwSecretKeyEnv             = "SCW_SECRET_KEY"
+	scwActiveProfileEnv         = "SCW_PROFILE"
+	scwApiUrlEnv                = "SCW_API_URL"
+	scwInsecureEnv              = "SCW_INSECURE"
+	scwDefaultOrganizationIDEnv = "SCW_DEFAULT_ORGANIZATION_ID"
+	scwDefaultRegionEnv         = "SCW_DEFAULT_REGION"
+	scwDefaultZoneEnv           = "SCW_DEFAULT_ZONE"
+
+	// All deprecated (cli&terraform)
+	terraformAccessKeyEnv    = "SCALEWAY_ACCESS_KEY" // used both as access key and secret key
+	terraformSecretKeyEnv    = "SCALEWAY_TOKEN"
+	terraformOrganizationEnv = "SCALEWAY_ORGANIZATION"
+	terraformRegionEnv       = "SCALEWAY_REGION"
+	cliTLSVerifyEnv          = "SCW_TLSVERIFY"
+	cliOrganizationEnv       = "SCW_ORGANIZATION"
+	cliRegionEnv             = "SCW_REGION"
+	cliSecretKeyEnv          = "SCW_TOKEN"
+
+	// TBD
+	cliVerboseEnv         = "SCW_VERBOSE_API"
+	cliDebugEnv           = "DEBUG"
+	cliNoCheckVersionEnv  = "SCW_NOCHECKVERSION"
+	cliTestWithRealAPIEnv = "TEST_WITH_REAL_API"
+	cliSecureExecEnv      = "SCW_SECURE_EXEC"
+	cliGatewayEnv         = "SCW_GATEWAY"
+	cliSensitiveEnv       = "SCW_SENSITIVE"
+	cliAccountApiEnv      = "SCW_ACCOUNT_API"
+	cliMetadataAPIEnv     = "SCW_METADATA_API"
+	cliMarketPlaceAPIEnv  = "SCW_MARKETPLACE_API"
+	cliComputePar1APIEnv  = "SCW_COMPUTE_PAR1_API"
+	cliComputeAms1APIEnv  = "SCW_COMPUTE_AMS1_API"
+	cliCommercialTypeEnv  = "SCW_COMMERCIAL_TYPE"
+	cliTargetArchEnv      = "SCW_TARGET_ARCH"
+)
+
+type Config interface {
+	GetAccessKey() string
+	GetSecretKey() string
+	GetApiUrl() string
+	GetInsecure() bool
+	GetDefaultOrganizationId() string
+	GetDefaultRegion() string
+	GetDefaultZone() string
+}
+
+type configV2 struct {
+	profile       `yaml:",inline"`
+	ActiveProfile *string             `yaml:"active_profile,omitempty"`
+	Profiles      map[string]*profile `yaml:"profiles,omitempty"`
+
+	withProfile string
```

Comment:
What does `whithProfile` exactly do/mean? I feel that its name could be more descriptive/concrete.